server.js

```diff
@@ -23,64 +35,107 @@ app.use(function (req, res, next) {
 });
 
 function render(req, res) {
-	if (req.text === undefined) {
-		res.status(400).send('The body must have Content-Type:text/html');
-		return;
-	}
+	if (req.is('application/json')) {
+		var headerHtml = req.body.header || '';
+		var bodyHtml = req.body.body;
+		var footerHtml = req.body.footer || '';			
+		var pageSize = req.body.pagesize|| 'A4';		
+		var pageLandscape = req.body.pagelandscape === 'true';
 
-	var html = req.text;
-	var format = req.query['format'] || 'pdf';
-	var pageSize = req.query['pagesize'] || 'A4';
-	var pageLandscape = req.query['pagelandscape'] === 'true';
-	var width = req.query['width'] || 210 * 4;
-	var height = req.query['height'] || 297 * 4;
-	var deviceScaleFactor = req.query['deviceScaleFactor'] || 1.0;
-
-	width = parseInt(width);
-	height = parseInt(height);
-	deviceScaleFactor = parseFloat(deviceScaleFactor);
-
-	(async () => {
-		try {
-			const browser = await puppeteer.launch(
-				{ args: ['--no-sandbox', '--disable-setuid-sandbox'] });
+		(async () => {
+			const browser = await puppeteer.launch({
+				args: ['--no-sandbox', '--disable-setuid-sandbox']
+			});
 			const page = await browser.newPage();
-			await page.setContent(html);
+			await page.setContent(bodyHtml);
 
-			if (format == 'pdf') {
+			if (headerHtml && footerHtml) {
 				await page.pdf({
 					format: pageSize,
 					landscape: pageLandscape,
-					printBackground: true
+					printBackground: true,
+					displayHeaderFooter: true,
```

Comment:
I feel this should be `displayHeaderFooter: headerHtml || footerHtml` then you don't need the `if`, and if we want to render something with only a header, or only a footer then this will still work